webrtc/src/data_channel/mod.rs

```diff
@@ -67,10 +72,10 @@ pub struct RTCDataChannel {
     // is created, the binaryType attribute MUST be initialized to the string
     // "blob". This attribute controls how binary data is exposed to scripts.
     // binaryType                 string
-    pub(crate) on_message_handler: Arc<Mutex<Option<OnMessageHdlrFn>>>,
-    pub(crate) on_open_handler: Arc<Mutex<Option<OnOpenHdlrFn>>>,
-    pub(crate) on_close_handler: Arc<Mutex<Option<OnCloseHdlrFn>>>,
-    pub(crate) on_error_handler: Arc<Mutex<Option<OnErrorHdlrFn>>>,
+    pub(crate) on_message_handler: Arc<ArcSwapOption<Mutex<OnMessageHdlrFn>>>,
+    pub(crate) on_open_handler: StdMutex<Option<OnOpenHdlrFn>>,
```

Comment:
and why there's no longer an Arc here? What if the user clones `DataChannel`?

Reply:
You mean `Arc<Mutex<Option<OnOpenHdlrFn>>>` => `StdMutex<Option<OnOpenHdlrFn>>`? Well i [refactored](https://www.diffchecker.com/E41ixG2e) a `RTCDataChannel::do_open` a bit, it used to clone this `Arc` into `tokio::spawn`, and now it takes the inner `OnMessageHdlrFn` from `Option` and moves it to that `tokio::spawn`.  I believe its ok, let me know if you want this reverted.